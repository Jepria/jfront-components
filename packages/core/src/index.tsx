export * from '@jfront-ui/app'
export * from '@jfront-ui/datepicker'
export * from '@jfront-ui/form'
//export * from '@jfront-ui/grid'
export * from '@jfront-ui/hooks'
//export * from '@jfront-ui/input'
export * from '@jfront-ui/layout'
//export * from '@jfront-ui/pagingbar'
//export * from '@jfront-ui/tabpanel'
//export * from '@jfront-ui/toolbar'
