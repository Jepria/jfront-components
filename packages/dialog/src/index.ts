export * from './SimpleDialog'
export * from './AlertDialog'
export * from './FormDialog'
export * from './ErrorDialog'
export * from './WarningDialog'