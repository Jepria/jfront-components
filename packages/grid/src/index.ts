export * from './common'
export * from './jepgrid'
