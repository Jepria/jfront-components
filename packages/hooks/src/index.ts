export * from "./useOnClickOutside"
export * from "./useRouteMapping"
export * from "./useTree/"
export * from "./useTree/types"
export * from "./useFilter"
