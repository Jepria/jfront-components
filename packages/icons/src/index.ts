export * from "./Add"
export * from "./Delete"
export * from "./Edit"
export * from "./Exclamation"
export * from "./First"
export * from "./Last"
export * from "./Loading"
export * from "./Next"
export * from "./OpenIcon"
export * from "./Prev"
export * from "./Refresh"
export * from "./Save"
export * from "./Search"
export * from "./SettingsSmall"
export * from "./Split"
export * from "./View"
