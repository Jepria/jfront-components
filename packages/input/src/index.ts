export interface InputProps {
  error?: string
  name?: string
  isLoading?: boolean
}

export * from "./SelectInput"
export * from "./TextInput"
export * from "./NumberInput"
export * from "./MaskedInput"
export * from "./MoneyInput"
export * from "./DecimalInput"
export * from "./MoneyInput"
