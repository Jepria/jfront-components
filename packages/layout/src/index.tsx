import styled from 'styled-components';

export const FormContainer = styled.div`
  padding: 5px;
`;

export const VerticalLayout = styled.div`
  width: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
`;

export const HorizontalLayout = styled.div`
  width: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
      -ms-flex-direction: row;
          flex-direction: row;
`;

export const Page = styled.div`
  box-sizing: border-box;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
`;

export const Header = styled.header`
  box-sizing: border-box;
  height: auto;
  width: 100%;
`;

export const Content = styled.section`
  box-sizing: border-box;
  width: 100%;
  overflow: auto;
  -webkit-box-flex: 1;
      -ms-flex: 1;
          flex: 1;
  -ms-flex-positive: 1;
      flex-grow: 1;
`;

export const Footer = styled.footer`
  box-sizing: border-box;
  height: auto;
  width: 100%;
`;
