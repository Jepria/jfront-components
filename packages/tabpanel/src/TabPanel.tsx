import styled from "styled-components"
import React from "react"

const TabPanel = styled.div`
  height: 22px;
  font-family: tahoma, arial, helvetica, sans-serif;
  color: rgb(21, 66, 139);
  font-size: 11px;
  z-index: 1;
`

interface TabProps {
  selected?: boolean
}

const Tab = styled.div<TabProps>`
  display: inline-block;
  height: 16px;
  text-align: center;
  margin-left: 2px;
  padding: 3px 6px 3px 6px;
  min-width: 20px;
  border: 1px solid #8db2e3;
  border-top-left-radius: 3px;
  border-top-right-radius: 3px;
  border-bottom-color: #d7e4f3;
  ${(props) =>
    props.selected
      ? `
  background-color: #D7E4F3;
    cursor: default;
    font-weight: bold;
  `
      : `
    background: linear-gradient(#dae6f4, #d0def0);
    cursor: pointer;
    &:hover{
    opacity: 0.8
  }
  `}
`

export { TabPanel, Tab }
