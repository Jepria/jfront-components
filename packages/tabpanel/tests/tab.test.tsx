import { render } from "@testing-library/react"
import React from "react"
import { Tab } from "../src"

test("Checking for the existence of an element Tab", () => {
  render(<Tab />)
})
