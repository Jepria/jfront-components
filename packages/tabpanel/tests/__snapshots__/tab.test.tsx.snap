// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot  1`] = `
<DocumentFragment>
  <div
    class="sc-AxirZ ercANe"
  />
</DocumentFragment>
`;
